### PR TITLE
Added Okta Verify enrollment with SMS and Email based activation

### PR DESCRIPTION
**Pull request by karthikaselvaraj-okta**
Added e2e test cases for Okta Verify enrollment with SMS and Email based activation

### CODE REVIEW

**Review by arvindkrishnakumar-okta — APPROVED** [review]
LGTM, please ensure the Bacon E2E tests run successfully before merging.

File: samples/embedded-auth-with-sdk/src/test/java/info/seleniumcucumber/userStepDefinitions/OktaVerify.java [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+	public OktaVerifyPage oktaVerifyPage = new OktaVerifyPage(driver);
+
+	public ClientBuilder builder = Clients.builder();
+	public Client client = builder.build();
+
+	private final Logger logger = LoggerFactory.getLogger(OktaVerify.class);
+
+
+	@When("^she selects Okta Verify from the list$")
+	public void She_selects_Okta_Verify_from_the_list() {
```
Comment (vijetmahabaleshwar-okta) [inline samples/embedded-auth-with-sdk/src/test/java/info/seleniumcucumber/userStepDefinitions/OktaVerify.java:52]: Rename all methods to have only lowercase letters. That's the naming convention we have used in other files.

Comment (karthikaselvaraj-okta) [reply samples/embedded-auth-with-sdk/src/test/java/info/seleniumcucumber/userStepDefinitions/OktaVerify.java:52]: Sure, will change it.

File: samples/embedded-auth-with-sdk/src/test/java/pages/OktaVerifyPage.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    public WebElement emailModeRadioButton;
+
+    @FindBy(css = "input[type='radio'][name='mode'][value='sms'][class='form-check-input']")
+    public WebElement smsModeRadioButton;
+
+    @FindBy(className="text-center")
+    public WebElement waitingScreen;
+
+    @FindBy(css = "input[type='text'][name='channelValue'][required='required']")
+    public WebElement textBox;
```
Comment (vijetmahabaleshwar-okta) [inline samples/embedded-auth-with-sdk/src/test/java/pages/OktaVerifyPage.java:44]: rename this to phoneOrEmailTextBox

File: samples/embedded-auth-with-sdk/src/test/java/pages/OktaVerifyPage.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+    @FindBy(className="okta-verify-desc-text")
+    public WebElement oktaVerifyScreen;
+
+    @FindBy(className="font-weight-light")
+    public WebElement authenticatorsOption;
```
Comment (vijetmahabaleshwar-okta) [inline samples/embedded-auth-with-sdk/src/test/java/pages/OktaVerifyPage.java:50]: This doesn't seem like a good selector for this element. `className="font-weight-light"` is too generic. Can you check if there's something more speciic to this element?

File: samples/embedded-auth-with-sdk/src/test/resources/features/okta_verify.feature [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    When she selects Okta Verify from the list
+    And  she sees the option "Can't scan"
+    When she clicks on "Can't scan"
+    Then she sees a list of modes to register
+    When she selects Email option
+    Then she sees a page to input the Email
+    And she fills out her Email for Okta verify
+    And she clicks on submit button saying "Send me the setup link"
+    Then the screen changes to a waiting screen saying "We sent an email with an Okta Verify setup link. To continue, open the link on your mobile device."
+    When she clicks the magic link in her email
```
Comment (vijetmahabaleshwar-okta) [inline samples/embedded-auth-with-sdk/src/test/resources/features/okta_verify.feature:61]: nit: Change this step to "she clicks on the link in her email"
This is not a magic link. It's just a download link.

File: samples/embedded-auth-with-sdk/src/test/java/pages/Page.java [hunk 5]
```diff
@@ -151,6 +151,30 @@ public String fetchMagicLinkFromEmail(String emailContent) {
         return matcher.find() ? matcher.group(1) : null;
     }
 
+    public String fetchOktaVerifyMagicLinkFromEmail(String emailContent) {
+        Pattern pattern = Pattern.compile("\\\"push-verify-activation-link\\\" href=\\\"(.*?)\\\"");
+        Matcher matcher = pattern.matcher(emailContent);
+        return matcher.find() ? matcher.group(1) : null;
+    }
+
+    public String fetchMagicLinkFromSMS() {
```
Comment (vijetmahabaleshwar-okta) [inline samples/embedded-auth-with-sdk/src/test/java/pages/Page.java:160]: Rename method to `fetchLinkFromSMS()` 
There's no magic link here.
Remove all references to magic.

File: samples/embedded-auth-with-sdk/src/test/java/pages/Page.java [hunk 6]
```diff
@@ -151,6 +151,30 @@ public String fetchMagicLinkFromEmail(String emailContent) {
         return matcher.find() ? matcher.group(1) : null;
     }
 
+    public String fetchOktaVerifyMagicLinkFromEmail(String emailContent) {
```
Comment (vijetmahabaleshwar-okta) [inline samples/embedded-auth-with-sdk/src/test/java/pages/Page.java:154]: Remove magic reference

**Review by vijetmahabaleshwar-okta — CHANGES_REQUESTED** [review]
Added a few comments which need to be addressed before merge.

File: samples/embedded-auth-with-sdk/src/test/java/pages/OktaVerifyPage.java (hunk 2 above)
Comment (karthikaselvaraj-okta) [reply samples/embedded-auth-with-sdk/src/test/java/pages/OktaVerifyPage.java:44]: Rename done

File: samples/embedded-auth-with-sdk/src/test/java/pages/OktaVerifyPage.java (hunk 3 above)
Comment (karthikaselvaraj-okta) [reply samples/embedded-auth-with-sdk/src/test/java/pages/OktaVerifyPage.java:50]: Have changed the element.

File: samples/embedded-auth-with-sdk/src/test/java/pages/Page.java (hunk 6 above)
Comment (karthikaselvaraj-okta) [reply samples/embedded-auth-with-sdk/src/test/java/pages/Page.java:154]: Changed all the magic link references

File: samples/embedded-auth-with-sdk/src/test/java/pages/Page.java (hunk 5 above)
Comment (karthikaselvaraj-okta) [reply samples/embedded-auth-with-sdk/src/test/java/pages/Page.java:160]: renamed the references

File: samples/embedded-auth-with-sdk/src/test/resources/features/okta_verify.feature (hunk 4 above)
Comment (karthikaselvaraj-okta) [reply samples/embedded-auth-with-sdk/src/test/resources/features/okta_verify.feature:61]: Done

**Review by vijetmahabaleshwar-okta — APPROVED** [review]

**Pull request merged**